### PR TITLE
support "nullable" types

### LINKED ISSUE #8: support "nullable" types
`null` is one of the unfortunate realities of JavaScript. In order to be useful in a broad range of settings, sanctuary-def should acknowledge this and enable the definition of types which include `null` as a member. `displayName :: Nullable String`, for example, states that `displayName` is either `null` or a string value.

It's currently possible to define `NullableString` via `$.NullaryType`:

```javascript
//    NullableString :: Type
const NullableString = $.NullaryType(
  'my-package/NullableString',
  x => x === null || $.String.test(x)
);

const def = $.create($.env.concat([NullableString]));

//    f :: NullableString -> NullableString
const f = def('f', {}, [NullableString, NullableString], x => x);
```

What about polymorphism? In Sanctuary, for example, we'd like [`S.toMaybe`][1] to have type `Nullable a -> Maybe a`. At first glance we already have a workable solution:

```javascript
//    Nullable :: Type -> Type
const Nullable = $.UnaryType(
  'my-package/Nullable',
  x => true,
  nullable => nullable === null ? [] : [nullable]
);
```

If `Nullable` exists in the environment, what are the types of which `'foo'` is a member? `String`, of course, but also `Nullable _` for some `_`. Resolving `_` is infinitely recursive because the "inner" value of a nullable is the nullable itself. We determine that `'foo' :: String` and `'foo' :: Nullable String`, but we also determine that `'foo' :: Nullable (Nullable String)`, `'foo' :: Nullable (Nullable (Nullable String))`, and so on.

sanctuary-def should provide a special `$.Nullable` type constructor.

[1]: https://github.com/plaid/sanctuary#toMaybe

### PR DESCRIPTION
**Pull request by davidchambers**
Closes #8

### CODE REVIEW

**Comment by svozza** [comment]
LGTM.

**Comment by davidchambers** [comment]
I must document this function before merging this pull request.

**Comment by davidchambers** [comment]
@svozza, does the README addition read well to you?

File: README.md [hunk 1]
```diff
@@ -433,6 +433,38 @@ dist(0);
...
+
+For example:
+
+```javascript
+//    toUpper :: Nullable String -> Nullable String
+const toUpper =
+def('toUpper',
+    {},
+    [$.Nullable($.String), $.Nullable($.String)],
+    R.unless(R.equals(null), R.toUpper));
```
Comment (svozza) [inline README.md]: Do we want the docs for `sanctuary-def` to assume a deep familiarity with Ramda? I use it every day and I still had to look up what `R.unless` does just to be sure I was reading the example properly.

Comment (davidchambers) [inline README.md]: Great point! Let's use a lambda instead.

Comment (davidchambers) [inline README.md]: :zap:

Comment (svozza) [inline README.md]: :+1:

**Pull request merged**